src/Administration/Resources/app/administration/src/module/sw-cms/service/cmsDataResolver.service.js

```diff
@@ -18,38 +18,46 @@ const slots = {};
 
 function resolve(page) {
     const loadedData = [];
-
-    contextService = Shopware.Context.api;
-    repoFactory = Shopware.Service('repositoryFactory');
-    cmsService = Shopware.Service('cmsService');
-    cmsElements = cmsService.getCmsElementRegistry();
-
     const slotEntityList = {};
-    page.sections.forEach((section) => {
-        section.blocks.forEach((block) => {
-            block.slots.forEach((slot) => {
-                slots[slot.id] = slot;
-                initSlotConfig(slot);
-                initSlotDefaultData(slot);
-
-                const slotData = cmsElements[slot.type].collect(slot);
-                if (Object.keys(slotData).length > 0) {
-                    slotEntityList[slot.id] = slotData;
-                }
+
+    try {
+        contextService = Shopware.Context.api;
+        repoFactory = Shopware.Service('repositoryFactory');
+        cmsService = Shopware.Service('cmsService');
+        cmsElements = cmsService.getCmsElementRegistry();
+
+        page.sections.forEach((section) => {
+            section.blocks.forEach((block) => {
+                block.slots.forEach((slot) => {
+                    slots[slot.id] = slot;
+                    initSlotConfig(slot);
+                    initSlotDefaultData(slot);
+                    const cmsElement = cmsElements[slot.type];
+
+                    if (cmsElement) {
+                        const slotData = cmsElement.collect(slot);
+                        if (Object.keys(slotData).length > 0) {
+                            slotEntityList[slot.id] = slotData;
+                        }
+                    } else {
+                        warn(`Missing registration for slot type ${slot.type}. Slot ${slot.id} Block ${block.name} (${block.id}) Section ${section.name} (${section.id})`);
+                    }
+                });
             });
         });
-    });
-
-    const { directReads, searches } = optimizeCriteriaObjects(slotEntityList);
 
-    loadedData.push(
-        fetchByIdentifier(directReads),
-    );
+        const { directReads, searches } = optimizeCriteriaObjects(slotEntityList);
 
-    loadedData.push(
-        fetchByCriteria(searches),
-    );
+        loadedData.push(
+            fetchByIdentifier(directReads),
+        );
 
+        loadedData.push(
+            fetchByCriteria(searches),
+        );
+    } catch (e) {
+        return Promise.resolve(e);
```

Comment:
Why are you resloving with the error?

Reply:
As said in 2. of the description

> Adds a try-catch to pass the caught exception as resolved promise back like the Promise.all().catch(). Unsure why it is like it is but I follow along. I would've rather seen a Promise.reject() there.

Look down a bit further down to the catch of the Promise.all. I am unsure why this is but I wanted to follow along so far.

Reply:
Guess I haven't read the description. well then Ill ask the component owner why its like that

Reply:
Thank you :)

Reply:
The component owner @tobiasberge  couldn't give me an answer, but if it was there before lets just re-add it

Reply:
re-add? I didn't remove stuff :/